src/Symfony/Component/Security/Http/Authenticator/AccessToken/AccessTokenAuthenticator.php

```diff
@@ -0,0 +1,119 @@
+     */
+    public function __construct(
+        private readonly UserProviderInterface $userProvider,
+        private readonly AccessTokenHandlerInterface $accessTokenHandler,
+        private readonly iterable $accessTokenExtractors,
+        private readonly ?AuthenticationSuccessHandlerInterface $successHandler = null,
+        AuthenticationFailureHandlerInterface $failureHandler = null,
+        string $realm = null,
+        LoggerInterface $logger = null
+    ) {
+        $this->logger = $logger ?? new NullLogger();
+        $this->failureHandler = $failureHandler ?? new DefaultAuthenticationFailureHandler($realm);
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return null !== $this->extractAccessToken($request);
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        $accessToken = $this->extractAccessToken($request);
+        if (null === $accessToken) {
+            throw new BadCredentialsException('Invalid credentials.');
+        }
+        $userIdentifier = $this->accessTokenHandler->getUserIdentifierFrom($accessToken);
+        $this->logger->debug('The access token was correctly handled', ['userIdentifier' => $userIdentifier]);
+
+        return new Passport(
+            new UserBadge($userIdentifier, $this->userProvider->loadUserByIdentifier(...)),
+            new AccessTokenCredentials($accessToken)
+        );
+    }
+
+    public function createToken(Passport $passport, string $firewallName): TokenInterface
+    {
+        return new AccessToken($passport->getUser(), $firewallName, $passport->getUser()->getRoles());
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
+    {
+        return $this->successHandler?->onAuthenticationSuccess($request, $token);
+    }
+
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception): ?Response
+    {
+        return $this->failureHandler->onAuthenticationFailure($request, $exception);
+    }
+
+    public function setTranslator(?TranslatorInterface $translator): void
+    {
+        if ($this->failureHandler instanceof DefaultAuthenticationFailureHandler) {
+            $this->failureHandler->setTranslator($translator);
+        }
+    }
+
+    private function extractAccessToken(Request $request): ?string
+    {
+        foreach ($this->accessTokenExtractors as $extractor) {
+            try {
+                $this->logger->debug('Trying to extract the access token', ['extractor' => $extractor]);
+                $accessToken = $extractor->extractAccessToken($request);
+                if (null !== $accessToken) {
+                    $this->logger->debug('An access token is present in the request');
+
+                    return $accessToken;
+                }
+            } catch (\Throwable $e) {
+                $this->logger->debug('An unexpected exception was thrown', ['exception' => $e]);
```

Comment:
Sorry, but this is at least an error, *if* we keep the catchall block. I'm still in favor of removing it.

First of all, if an extractor throws, it is broken and should be fixed. If we swallow the error, we basically tolerate broken extractor implementations which we shouldn't.

Secondly, if I have more than one way to extract a token and I have configured priorities among them, an error I did not anticipate could mean that this priority is bypassed. This means my second extractor is called in situations where I really wanted my first extractor to take over. That sounds dangerous to me.

Reply:
I'm against catching `\Throwable` too.

Reply:
OK. I reverted back to the previous behaviour.
I also added a test to chack that, in the absence of the access token, anonymous access is still possible when an access control rule exists.